pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -1552,13 +1552,14 @@ func initPostOnboardSubs(zedagentCtx *zedagentContext) {
 	}
 
 	zedagentCtx.subAppDiskMetric, err = ps.NewSubscription(pubsub.SubscriptionOptions{
-		AgentName:   "volumemgr",
-		MyAgentName: agentName,
-		TopicImpl:   types.AppDiskMetric{},
-		Activate:    true,
-		Ctx:         zedagentCtx,
-		WarningTime: warningTime,
-		ErrorTime:   errorTime,
+		AgentName:     "volumemgr",
+		MyAgentName:   agentName,
+		TopicImpl:     types.AppDiskMetric{},
+		Activate:      true,
+		Ctx:           zedagentCtx,
+		CreateHandler: handleAppDiskMetricCreate,
```

Comment:
Do we care about getting modifications/updates to the AppDiskMetric? Or we just need an initial set of metrics?

Reply:
We have metric message to send metrics of volume constantly after created. But at the same time we have `info.VolumeResources` field to send such metric as part of info message. My understanding here that we should fill only initial data in info, that is max size and current size during volume creation. So it is better to fill the data from handleVolumeStatusImpl if we already have AppDiskMetric (as we do it now) or from CreateHandler of AppDiskMetric subscription if we already have VolumeStatus (which I add to the PR).

Reply:
OK